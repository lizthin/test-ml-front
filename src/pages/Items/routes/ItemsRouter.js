import React from 'react';
import {
    Switch,
    Route,
} from "react-router-dom";
import Header from '../../../components/Header/Header';
import ShowDetail from '../components/ShowDetail/ShowDetail';
import ShowItems from '../components/ShowItems/ShowItems';

import PropTypes from 'prop-types'

/**
 * funtion component ItemsRouter, call component with change path 
 * @param {props object} porps parentHistory, route props 
 * @returns component route 
 */
const  ItemsRouter = ({parentHistory}) => {
        return (
            <>
                <Header parentHistory={parentHistory}/> 
                <div>
                    <Switch>
                        <Route exact path="/items" component={ShowItems} ></Route>
                        <Route exact path="/items/:id" component={ShowDetail} ></Route>
                    </Switch>

                </div>

            </>
        );
};
ItemsRouter.propTypes = {
    parentHistory: PropTypes.object.isRequired
}
export default ItemsRouter;