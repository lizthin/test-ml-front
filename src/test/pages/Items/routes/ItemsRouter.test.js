import React from 'react';
import { shallow } from 'enzyme';
import ItemsRouter from '../../../../pages/Items/routes/ItemsRouter';


describe('Test ItemsRouter', () => {
    test('should display ItemsRouter', () => {
        const wrapper = shallow(<ItemsRouter parentHistory={{ history: {} }} />);
        expect(wrapper).toMatchSnapshot();
    })
})