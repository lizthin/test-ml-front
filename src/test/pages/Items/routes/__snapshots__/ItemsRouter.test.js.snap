// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ItemsRouter should display ItemsRouter 1`] = `
<Fragment>
  <Header
    parentHistory={
      Object {
        "history": Object {},
      }
    }
  />
  <div>
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/items"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/items/:id"
      />
    </Switch>
  </div>
</Fragment>
`;
